RogueEssence/Menu/Settings/SettingsTitleMenu.cs

```diff
@@ -0,0 +1,265 @@
+﻿using NLua;
+using RogueElements;
+using RogueEssence.Content;
+using RogueEssence.Data;
+using RogueEssence.Dungeon;
+using RogueEssence.Ground;
+using System;
+using System.Collections.Generic;
+using System.Data;
+using System.Globalization;
+using System.Linq;
+
+namespace RogueEssence.Menu
+{
+    public class SettingData
+    {
+        public string Name;
+        public List<string> Options;
+        public int Default;
+        public Action<MenuSetting> SettingChangeAction;
+        public Action<MenuSetting> SaveAction;
+        public SettingData(string name, string[] options, int defaultValue, Action<MenuSetting> saveAction, Action<MenuSetting> settingChangeAction = null) : this(name, options.ToList(), defaultValue, saveAction, settingChangeAction) { }
+        public SettingData(string name, List<string> options, int defaultValue, Action<MenuSetting> saveAction, Action<MenuSetting> settingChangeAction = null)
+        {
+            Name = name;
+            Options = options;
+            Default = defaultValue;
+            SaveAction = saveAction;
+            SettingChangeAction = settingChangeAction;
+        }
+    }
+    public class SettingsPage
+    {
+        public string Title = "";
+        public List<SettingData> Choices = new();
+        public Action GlobalSaveAction;
+        public SettingsPage(string title)
+        {
+            Title = title;
+        }
+        public void AddSetting(string name, LuaTable options, int defaultValue, Action<MenuSetting> action, Action<MenuSetting> settingChangedAction = null)
+        {
+            Object[] arr = new Object[options.Values.Count];
+            options.Values.CopyTo(arr, 0);
+            List<string> list = arr.Select(a => a.ToString()).ToList();
+            AddSetting(name, list, defaultValue, action, settingChangedAction);
+        }
+        public void AddSetting(string name, string[] options, int defaultValue, Action<MenuSetting> action, Action<MenuSetting> settingChangedAction = null)
+        {
+            AddSetting(name, options.ToList(), defaultValue, action, settingChangedAction);
+        }
+        public void AddSetting(string name, List<string> options, int defaultValue, Action<MenuSetting> action, Action<MenuSetting> settingChangedAction = null)
+        {
+            Choices.Add(new SettingData(name, options, defaultValue, action, settingChangedAction));
+        }
+    }
+
+    public class SettingsTitleMenu : InteractableMenu
+    {
+        private static string _originId = "origin";
```

Comment:
Use PathMod.BaseNamespace for the namespace of the base game.

Reply:
Done with the last commit